doc/network.md

```diff
@@ -0,0 +1,325 @@
+| N-1    | A        | The N-1 element of the array if present.    |
+
+### Tuple `(A, B, ...)`
+
+A tuple is a type which holds the content of multiple types specified in its name.
+The tuple type is very abstract and acts as many fields as the tuple should hold.
+
+Size: size of `A` + size of `B` + ...
+
+| Fields | Type | Description                    |
+|:------:|:----:|:-------------------------------|
+| first  | `A`  | The first value of the tuple.  |
+| second | `B`  | The second value of the tuple. |
+| ...    | ...  | Another value of the tuple.    |
+
+### Optional `X?`
+
+An optional is a value which may or may not be present.
+
+Size: 0 if non-present, else size of `X`.
+
+## Packets
+
+### Workspace-related Packets
+
+The server controls most of the requests sent by a client for workspace-related packets as it requires some permissions.
+Those packets are most likely sent only on build servers.
+
+#### `plasmid:workspace/enter`
+
+Packet sent when a player enters a workspace editor. The client should render the bounds of the map if present,
+and the contained regions, points, etc.  
+A `plasmid:workspace/regions` packet should be sent with this packet if the workspace contains regions.
+
+Direction: `C<-S`
+
+| Fields | Type       | Description                                                            |
+|:------:|:----------:|:-----------------------------------------------------------------------|
+| id     | Identifier | The identifier of the entered workspace.                               |
+| bounds | [Bounds]   | The bounds of the map, may be `[[0, 0, 0], [0, 0, 0]]` if not present. |
+| world  | Identifier | The world in which the workspace is present.                           |
+| data   | NBT Tag    | Arbitrary data assigned to the map template.                           |
+
+#### `plasmid:workspace/bounds`
+
+Sets the bounds of the workspace map. The client should render those bounds if non-null.
+A client can request bounds change, but can be rejected. If a request is accepted, the packet is sent back to the client.
+
+Direction `C<->S`
+
+| Fields | Type       | Description                                                                      |
+|:------:|:----------:|:---------------------------------------------------------------------------------|
+| id     | Identifier | The identifier of the affected workspace.                                        |
+| bounds | [Bounds]   | The new bounds of the workspace map, `[[0, 0, 0], [0, 0, 0]]` if bounds is null. |
+
+#### `plasmid:workspace/data`
+
+Sets the associated data on the workspace map.
+A client can request data change, but can be rejected. If a request is accepted, the packet is sent back to the client.
+
+Direction: `C<->S`
+
+| Fields | Type       | Description                               |
+|:------:|:----------:|:------------------------------------------|
+| id     | Identifier | The identifier of the affected workspace. |
```

Comment:
This is sometimes referred to as just `id` and sometimes as `workspace`- maybe we should standardise them all as `workspace`? Or `workspace_id` if we need to be more specific